src/gc.c

```diff
@@ -3205,9 +3205,11 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
     JL_PROBE_GC_SWEEP_BEGIN(sweep_full);
     {
         JL_TIMING(GC, GC_Sweep);
+        JL_COND_TIMING(sweep_full, GC, GC_FullSweep);
+        JL_COND_TIMING(!sweep_full, GC, GC_IncrementalSweep);
 #ifdef USE_TRACY
         if (sweep_full) {
-            TracyCZoneCtx ctx = *(JL_TIMING_CURRENT_BLOCK->tracy_ctx);
+            TracyCZoneCtx ctx = *(JL_TIMING_BLOCK(GC, GC_Sweep)->tracy_ctx);
```

Comment:
This is my first time looking at the new events rather than the API changes, and nesting these events here is confusing to me, especially considering that in Tracy we end up coloring the parent event based on the child even it's given.

IMO we want a runtime-selected `event` here rather than a runtime `enable/disable` per-event (although that will still probably be useful for other cases).

What do you think about tweaking the `JL_TIMING` to have a variant that accepts runtime-chosen owner/event?

```c
jl_timing_event_t event = jl_timing_event_gc_incremental_sweep;
if (sweep_full)
    event = jl_timing_event_gc_full_sweep;

JL_TIMING_(jl_timing_owner_gc, event);
```

Reply:
Do disabled events still show up in Tracy? A JL_SELECT_TIMING macro would be possible, but we could also just skip marking the sweep event and that way we don't have two of the same event?

Reply:
Maybe some inspiration can be gleaned from https://www.intel.com/content/www/us/en/docs/vtune-profiler/user-guide/2023-0/task-api.html

We currently the event API, but we could also use the task&domain API which would give use
"Domain enable/disable" https://www.intel.com/content/www/us/en/docs/vtune-profiler/user-guide/2023-0/domain-api.html

And potentially better nesting, albeit I haven't seen that used in the wild.